### PR TITLE
Add Swift to OpenStackControlPlane

### PR DESCRIPTION
**Pull request by cschwede**
This allows deploying the Swift operator and also updates the default configuration samples to deploy a single replica Swift storage backend.

### CODE REVIEW

**Comment by abays** [comment]
> @cschwede: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> Test name 	Commit 	Details 	Required 	Rerun command
> ci/prow/openstack-operator-build-deploy 	[2ed31fb](https://github.com/openstack-k8s-operators/openstack-operator/commit/2ed31fb36c1ba3cf4e0d2ade78d70d390c1a9bda) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/352/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1671098647317581824) 	true 	`/test openstack-operator-build-deploy`
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openstack-k8s-operators&repo=openstack-operator&pr=352). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:cschwede).

It looks like the deployment timed-out because the Swift resources never become `Ready`:

```
    - lastTransitionTime: "2023-06-20T10:55:35Z"
      message: OpenStackControlPlane Swift in progress
      reason: Requested
      severity: Info
      status: "False"
      type: OpenStackControlPlaneSwiftReady
```

(https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/352/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1671098647317581824/artifacts/openstack-operator-build-deploy/openstack-k8s-operators-gather/artifacts/api-resources/openstackcontrolplanes.yaml)

Looking at the `Swift` CR itself, you can see that its `Ready` condition is indeed not `True`:

```
    - lastTransitionTime: "2023-06-20T10:55:35Z"
      message: Setup started
      reason: Requested
      status: Unknown
      type: Ready
```

(https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/352/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1671098647317581824/artifacts/openstack-operator-build-deploy/openstack-k8s-operators-gather/artifacts/api-resources/swifts.yaml)

I looked at the Swift operator code itself and noticed that it does not follow the typical reconcile pattern we have used in the other Podified operators.  But the particular issue here is that it never sets the `Ready` condition other than to `Unknown` during initialization.  We'll need to update the operator to at least set this to `True` when appropriate and then perhaps follow-up beyond that to make the operator conform to the Podified reconcile pattern in general.

**Comment by cschwede** [comment]
The Swift resource should be fine now, according to https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/352/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1671204011975380992/artifacts/openstack-operator-build-deploy/openstack-k8s-operators-gather/artifacts/api-resources/swifts.yaml

However, it looks like Cinder is now failing: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_openstack-operator/352/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy/1671204011975380992/artifacts/openstack-operator-build-deploy/openstack-k8s-operators-gather/artifacts/api-resources/cinders.yaml

The [openstack-operator-crc-podified-edpm-deployment](https://review.rdoproject.org/zuul/build/1dd827eb14d645ab9b1fa09f10891284) because SwiftStorage tries to setup a PVC using the storageClass "local-storage" and it looks like only crc-csi-hostpath-provisioner is available.

File: apis/core/v1beta1/openstackcontrolplane_webhook.go [hunk 1]
```diff
@@ -186,6 +190,13 @@ func (r *OpenStackControlPlane) ValidateServices(basePath *field.Path) field.Err
 		}
 	}
 
+	if r.Spec.Swift.Enabled {
+		if depErrorMsg := r.checkDepsEnabled("Swift"); depErrorMsg != "" {
+			err := field.Invalid(basePath.Child("cinder").Child("enabled"), r.Spec.Swift.Enabled, depErrorMsg)
```
Comment (abays) [inline apis/core/v1beta1/openstackcontrolplane_webhook.go:195]: `s/cinder/swift`

Comment (cschwede) [reply apis/core/v1beta1/openstackcontrolplane_webhook.go]: Done

File: go.mod [hunk 2]
```diff
@@ -34,6 +34,8 @@ require (
 	sigs.k8s.io/controller-runtime v0.14.6
 )
 
+require github.com/openstack-k8s-operators/swift-operator/api v0.0.0-20230620193035-7e49c6696cbf
```
[the remark this answers is not in the document]
Comment (cschwede) [reply go.mod]: Done

**Comment by abays** [comment]
recheck

**Comment by cschwede** [comment]
recheck

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**